aws/delta-checkpoint/src/main.rs

```diff
@@ -0,0 +1,201 @@
+//! Lambda function for writing checkpoints from S3 event triggers on _delta_log JSON files.
+//!
+
+// Reference:
+//
+// https://aws.amazon.com/blogs/opensource/rust-runtime-for-aws-lambda
+// https://docs.aws.amazon.com/cli/latest/reference/s3api/put-bucket-notification-configuration.html
+// https://docs.aws.amazon.com/lambda/latest/dg/with-s3.html
+// https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-how-to-filtering.html
+
+use deltalake::checkpoints::{CheckPointWriter, CheckPointWriterError};
+use deltalake::DeltaDataTypeVersion;
+use lambda_runtime::{handler_fn, Context, Error};
+use lazy_static::lazy_static;
+use log::*;
+use regex::Regex;
+use serde_json::Value;
+use std::num::ParseIntError;
+use std::path::PathBuf;
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let func = handler_fn(func);
+    lambda_runtime::run(func).await?;
+    Ok(())
+}
+
+async fn func(event: Value, _: Context) -> Result<(), CheckPointLambdaError> {
+    process_event(&event).await
+}
+
+async fn process_event(event: &Value) -> Result<(), CheckPointLambdaError> {
+    let (bucket, key) = bucket_and_key_from_event(&event)?;
+    let (path, version) = table_path_and_version_from_key(key.as_str())?;
+    let table_uri = table_uri_from_parts(bucket.as_str(), path.as_str())?;
+
+    if version % 10 == 0 && version != 0 {
```

Comment:
I assume this is just to replicating the behavior in the Spark checkpoint writer? Assuming so, I think a comment along those lines here, and a mention in the README would be beneficial

Reply:
Comment added

```
    // Checkpoints are created for every 10th delta log commit.
    // Follows the reference implementation described in the delta protocol doc.
    // https://github.com/delta-io/delta/blob/master/PROTOCOL.md#checkpoints.

```